test/built-ins/Array/prototype/pop/clamps-to-integer-limit.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2017 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.pop
+description: >
+  Length values exceeding 2^53-1 are clamped to 2^53-1.
+info: |
+  1. ...
+  2. Let len be ? ToLength(? Get(O, "length")).
+  ...
+  4. Else len > 0,
+  a. Let newLen be len-1.
+  ...
+  e. Perform ? Set(O, "length", newLen, true).
+  ...
+---*/
+
+var arrayLike = {};
+
+arrayLike.length = 2**53 - 1;
+Array.prototype.pop.call(arrayLike);
+assert.sameValue(arrayLike.length, 2**53 - 2, "Length is 2**53 - 1");
```

Comment:
If _Length values exceeding 2^53-1 are clamped to 2^53-1._,  the expected value should be `2 ** 53 - 1`, right?

but wait...

This is clamping everything to `2**53-2`! Is that intended? cc @bterlson 

We should remove the `a. Let newLen be len-1.` after using `ToLength`, IMHO.